File: src/main/java/com/github/thorben/webnd/domain/user/FirstName.java
package com.github.thorben.webnd.domain.user;

import com.github.thorben.webnd.foundation.AbstractValueObject;

import javax.persistence.Embeddable;

@Embeddable
public class FirstName extends AbstractValueObject<String> {

	protected FirstName() {
		// JPA
	}

	public FirstName(String value) {
		super(value);
	}
}

File: src/main/kotlin/com/github/thorben/webnd/domain/user/FirstName.kt
package com.github.thorben.webnd.domain.user

import com.github.thorben.webnd.foundation.AbstractValueObject
import javax.persistence.Embeddable

@Embeddable
class FirstName(value: String) : AbstractValueObject<String>(value) {
	protected constructor(): this("") {
		// JPA
	}
}